routes/users.js

```diff
@@ -98,4 +98,56 @@ usersRouter.get('/:userId/orders', requireUser, async (req,res,next) => {
   }
 })
 
+usersRouter.patch('/:userId', requireUser, async (req, res, next) => { 
+    const { firstName, lastName, email, address, city, state, zip, isAdmin, username, password} = req.body;
+    const { userId } = req.params;
+
+    const updateFields = {}; 
+
+    if (firstName){
```

Comment:
Not required, but would make it much more elegant if we added all the fields at once. Something like this (note, un-tested code. just an idea...)
```js
const fieldsToUpdate = {firstName: true, lastName: true, email: true, /*etc*/};
for(let key of Object.keys(req.body)) {
  if (fieldsToUpdate[key]) {
    updateFields[key] = req.body[key];
  }
}
```

Reply:
I'll look into this. Thank you.